builtin/providers/dns/data_dns_a_record.go

```diff
@@ -0,0 +1,77 @@
+package dns
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"net"
+	"sort"
+)
+
+func dataSourceDnsARecord() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDnsARecordRead,
+		Schema: map[string]*schema.Schema{
+			"host": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			// Optionally sort A records in alphabetical order.
+			// This is helpful when a name uses round-robin DNS, which may
+			// sort records with multiple addresses in a non-deterministic order.
+			// This random sorting can cause flapping in terraform plans, where
+			// the changes in sort order cause dependent resources to update
+			// despite having no real change in the set of addresses.
+			"sort": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			// Optionally filter IPv6 records from DNS replies.
+			// This is helpful for other resources that do no support IPv6 yet,
+			// such as AWS security groups
+			"ipv4": &schema.Schema{
```

Comment:
Technically speaking, `A` records are only IPV4, and `AAAA` records are IPV6. `net.LookupIP` unfortunately looks like it does both, so should we do the filtering always? Some food for thought.